### PR TITLE
fix: bugs

### PR DESCRIPTION
**Pull request by akshita-sukhlecha**
## Changes 

1. Remove usage of langs field from 5 tasks. Replace self.langs to self.hf_subsets. 

2. Correct link of WikipediaRetrievalMultilingual task 

3. MLQARetrieval tasks : Corpus has duplicate contexts. We must deduplicate them at the time of data loading . 

## Checklist
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`.

### CODE REVIEW

File: docs/tasks.md
Comment (KennethEnevoldsen) [inline docs/tasks.md:1]: This is autogenerated - but you also changed the source so it is fine

File: mteb/tasks/Classification/multilingual/AfriSentiClassification.py [hunk 1]
```diff
@@ -65,7 +65,7 @@ def load_data(self, **kwargs):
         if self.data_loaded:
             return
         self.dataset = {}
-        for lang in self.langs:
+        for lang in self.hf_subsets:
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/multilingual/AfriSentiClassification.py:68]: ahh thanks good catch - I did the PR that changes this - should be more careful in the future.

File: mteb/tasks/Retrieval/multilingual/MLQARetrieval.py [hunk 2]
```diff
@@ -132,31 +132,32 @@ def load_data(self, **kwargs):
...
+            self.corpus[lang_pair] = {}
+            self.relevant_docs[lang_pair] = {}
 
-            self.corpus[lang_pair] = {
-                eval_split: {
-                    str(row["id"]): row for row in _dataset_raw[lang_pair][eval_split]
-                }
-                for eval_split in self.metadata_dict["eval_splits"]
-            }
+            for eval_split in self.metadata_dict["eval_splits"]:
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Retrieval/multilingual/MLQARetrieval.py:140]:
```suggestion
            for eval_split in self.metadata.eval_splits:
```

we are trying to move away from using the dict.

**Review by KennethEnevoldsen — COMMENTED** [review]
@imenelydiaker I believe it is fine to change MLQA since it hasn't been used in any benchmarks yet or should we create a new version?

**Comment by imenelydiaker** [comment]
> @imenelydiaker I believe it is fine to change MLQA since it hasn't been used in any benchmarks yet or should we create a new version?
I'm fine with this optimization, loading the context once for the same language seems a good solution. We can keep it in the same file I guess.

@akshita-sukhlecha Can you please make sure that thsi implementation recovers the same results as previous ones?

**Comment by akshita-sukhlecha** [comment]
@imenelydiaker The MLQA change is not for optimization, but to rectify the dataset loading. 
There are duplicate passages/contexts present across the rows. But they are not being deduplicated. 

For eg. in the **test** split of **mlqa.en.en**, there are total 11590 rows with 11582 unique questions and only 9916 unique passages. 
(I got these numbers from the huggingface dataset, but we can also check it here -[kaggle_link](https://www.kaggle.com/datasets/thedevastator/mlqa-multilingual-question-answering-dataset/data?select=mlqa.en.en_test.csv) )
But a total of 11590 passages are being added to the corpus currently, each with unique id.

So, I have de-duplicated the passages. The results would differ now as the old results were incorrect.

**Comment by akshita-sukhlecha** [comment]
We also need to de-duplicate the questions. But duplication in questions is very less. 
I will also push a commit for this in a day.

File: docs/mmteb/points/898.jsonl [hunk 3]
```diff
@@ -0,0 +1 @@
+{"GitHub": "akshita-sukhlecha", "Bug fixes": 2}
```
Comment (KennethEnevoldsen) [inline docs/mmteb/points/898.jsonl:1]:
```suggestion
{"GitHub": "akshita-sukhlecha", "Bug fixes": 2}
{"GitHub": "KennethEnevoldsen", "Review PR": 2}
{"GitHub": "imenelydiaker", "Review PR": 2}
```

File: docs/tasks.md [hunk 4]
```diff
@@ -513,7 +513,7 @@ The following tables give you an overview of the tasks in MTEB.
...
-| [WikipediaRetrievalMultilingual](https://huggingface.co/datasets/ellamind/wikipedia-2023-11-retrieval-pt) | ['ben', 'bul', 'ces', 'dan', 'deu', 'eng', 'fas', 'fin', 'hin', 'ita', 'nld', 'nor', 'por', 'ron', 'srp', 'swe'] | Retrieval | s2p | [Encyclopaedic] | {'en': 1500, 'de': 1500, 'it': 1500, 'pt': 1500, 'nl': 1500, 'cs': 1500, 'ro': 1500, 'bg': 1500, 'sr': 1500, 'fi': 1500, 'da': 1500, 'fa': 1500, 'hi': 1500, 'bn': 1500, 'no': 1500, 'sv': 1500} | {'test': 452.0} |
+| [WikipediaRetrievalMultilingual](https://huggingface.co/datasets/ellamind/wikipedia-2023-11-retrieval-multilingual-queries) | ['ben', 'bul', 'ces', 'dan', 'deu', 'eng', 'fas', 'fin', 'hin', 'ita', 'nld', 'nor', 'por', 'ron', 'srp', 'swe'] | Retrieval | s2p | [Encyclopaedic] | {'en': 1500, 'de': 1500, 'it': 1500, 'pt': 1500, 'nl': 1500, 'cs': 1500, 'ro': 1500, 'bg': 1500, 'sr': 1500, 'fi': 1500, 'da': 1500, 'fa': 1500, 'hi': 1500, 'bn': 1500, 'no': 1500, 'sv': 1500} | {'test': 452.0} |
```
Comment (KennethEnevoldsen) [inline docs/tasks.md:516]: this table is auto-updated

**Comment by KennethEnevoldsen** [comment]
> We also need to de-duplicate the questions. But duplication in questions is very less.
I will also push a commit for this in sometime.

@akshita-sukhlecha, as we are changing the results of a task, I would much rather do that in one go.

**Comment by KennethEnevoldsen** [comment]
Enabling auto-merge on this PR. Changing task results for a talk that is currently unused in benchmark seems reasonable, especially when they are duplicates

**Comment by KennethEnevoldsen** [comment]
#940 fixes tests

**Comment by akshita-sukhlecha** [comment]
> > We also need to de-duplicate the questions. But duplication in questions is very less.
> > I will also push a commit for this in sometime.
> 
> @akshita-sukhlecha, do you have the time to make this PR? Then I can review asap

I had pushed the commit for these changes in this PR itself : 
https://github.com/embeddings-benchmark/mteb/pull/898/commits/a43a19a5f96ead519612ff6b4fa4f24ebb1dd03d 

So, this is done !

**Pull request merged**